File: aqa_js/automation_playwright/pom/contactdetailspage.js
const {Page, expect} = require('@playwright/test');
const BasePage = require('./basepage');

class ContactDetailsPage extends BasePage {
    /**
     * @param {Page} page - Playwright Page object
     */
    constructor(page) {
        super(page);
        this.PageUrl = `${this.config.BASE_URL}/contactDetails`; // Replace with the actual URL of the signup page
        this.ids = {
            editContactButtonId: "#edit-contact",
            deleteContactButtonId: "#delete",
            returnButtonId: "#return",
            logoutButtonId: "#logout",
            contactDetailsFormId: "#contactDetails",
            firstNameFieldId: "#firstName",
            lastNameFieldId: "#lastName",
            emailFieldId: "#email",
            birthdateFieldId: "#birthdate",
            phoneFieldId: "#phone",
            street1AddressFieldId: "#street1",
            street2AddressFieldId: "#street2",
            cityFieldId: "#city",
            stateFieldId: "#stateProvince",
            postalCodeFieldId: "#postalCode",
            countryFieldId: "#country"
        };
    }

    async navigate() {
        await super.go_to_url(this.PageUrl);
    }
    async click_logout() {
        await this.page.click(this.ids.logoutButtonId);
    }
    async click_edit_contact() {
        await this.page.click(this.ids.editContactButtonId);
    }   
    async click_delete_contact() {
        await this.page.click(this.ids.deleteContactButtonId);
    }   
    async click_back_to_contact_list() {
        await this.page.click(this.ids.returnButtonId);
    }
    async get_first_name(){
        return await this.page.textContent(this.ids.firstNameFieldId)
    }
    async get_last_name(){
        return await this.page.textContent(this.ids.lastNameFieldId)
    }
    async get_email(){
        return await this.page.textContent(this.ids.emailFieldId)
    }
    async get_birthdate(){
        return await this.page.textContent(this.ids.birthdateFieldId)
    }
    async get_phone(){
        return await this.page.textContent(this.ids.phoneFieldId)
    }
    async get_street1(){
        return await this.page.textContent(this.ids.street1AddressFieldId)
    }
    async get_street2(){
        return await this.page.textContent(this.ids.street2AddressFieldId)
    }
    async get_city(){
        return await this.page.textContent(this.ids.cityFieldId)
    }
    async get_state(){
        return await this.page.textContent(this.ids.stateFieldId)
    }
    async get_postal_code(){
        return await this.page.textContent(this.ids.postalCodeFieldId)
    }
    async get_country(){
        return await this.page.textContent(this.ids.countryFieldId)
    }
    async confirm_alert() {
        const alert = await this.driver.switchTo().alert();
        await alert.accept();
    }

}
module.exports = ContactDetailsPage;

File: aqa_python/automation_playwright/pom/contactdetails_page.py
from pom.base_page import BasePage

class ContactDetailsPage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.page_url = f"{self.config.BASE_URL}/contactDetails"  # Update if necessary
        self.selectors = {
            "edit_contact_button": "#edit-contact",
            "delete_contact_button": "#delete",
            "return_button": "#return",
            "logout_button": "#logout",
            "contact_details_form": "#contactDetails",
            "first_name": "#firstName",
            "last_name": "#lastName",
            "email": "#email",
            "birthdate": "#birthdate",
            "phone": "#phone",
            "street1": "#street1",
            "street2": "#street2",
            "city": "#city",
            "state": "#stateProvince",
            "postal_code": "#postalCode",
            "country": "#country"
        }

    def navigate(self):
        self.go_to_url(self.page_url)
        self.page.wait_for_timeout(200)

    def click_logout(self):
        self.page.click(self.selectors["logout_button"])
        self.page.wait_for_timeout(200)

    def click_edit_contact(self):
        self.page.click(self.selectors["edit_contact_button"])
        self.page.wait_for_timeout(200)

    def click_delete_contact(self):
        self.page.click(self.selectors["delete_contact_button"])
        self.page.wait_for_timeout(200)

    def click_back_to_contact_list(self):
        self.page.click(self.selectors["return_button"])
        self.page.wait_for_timeout(200)

    def get_first_name(self) -> str:
        return self.page.text_content(self.selectors["first_name"])

    def get_last_name(self) -> str:
        return self.page.text_content(self.selectors["last_name"])

    def get_email(self) -> str:
        return self.page.text_content(self.selectors["email"])

    def get_birthdate(self) -> str:
        return self.page.text_content(self.selectors["birthdate"])

    def get_phone(self) -> str:
        return self.page.text_content(self.selectors["phone"])

    def get_street1(self) -> str:
        return self.page.text_content(self.selectors["street1"])

    def get_street2(self) -> str:
        return self.page.text_content(self.selectors["street2"])

    def get_city(self) -> str:
        return self.page.text_content(self.selectors["city"])

    def get_state(self) -> str:
        return self.page.text_content(self.selectors["state"])

    def get_postal_code(self) -> str:
        return self.page.text_content(self.selectors["postal_code"])

    def get_country(self) -> str:
        return self.page.text_content(self.selectors["country"])

    def confirm_alert(self):
        self.page.on("dialog", lambda dialog: dialog.accept())
